src/app/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import {Product} from "../model/product";
+import {ActivatedRoute} from "@angular/router";
+import {ProductsService} from "../services/products.service";
+import { Location } from '@angular/common';
+import {OrderService} from "../services/order.service";
+import {LoginService} from "../services/login.service";
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit {
+
+  constructor(private route: ActivatedRoute, private productService: ProductsService, private location: Location, private orderService: OrderService, private  loginService: LoginService) { }
+
+  product : Product | undefined;
+
+  isAdmin = this.loginService.isAdmin();
+
+  isCostumer = this.loginService.isCostumer();
+
+  ngOnInit(): void {
+    this.getProduct();
+  }
+
+  getProductId()
+  {
+    return Number(this.route.snapshot.paramMap.get('id'));
+  }
+
+  getProduct(): void {
+    this.productService.getProductById(this.getProductId())
+      .subscribe((product) => {this.product = product;},
```

Comment:
**suggestion:** take care to avoid memory leaks in your code by unsubscribing from observables. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/